From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: booky10 <boooky10@gmail.com>
Date: Sun, 16 Jul 2023 20:32:10 +0200
Subject: [PATCH] Fix non-adventure components not being translated

If an entity ever gets unloaded, this destroys every translation system

diff --git a/src/main/java/io/papermc/paper/adventure/PaperAdventure.java b/src/main/java/io/papermc/paper/adventure/PaperAdventure.java
index f2f4575e6ec4f5a90b8f48306b0edfef2e5527f5..5ec4339fde101cbf1380dc7104f6d7894e8b9031 100644
--- a/src/main/java/io/papermc/paper/adventure/PaperAdventure.java
+++ b/src/main/java/io/papermc/paper/adventure/PaperAdventure.java
@@ -147,6 +147,12 @@ public final class PaperAdventure {
     // Component
 
     public static Component asAdventure(final net.minecraft.network.chat.Component component) {
+        // CloudPlane start
+        if (component instanceof AdventureComponent) {
+            return ((AdventureComponent) component).adventure;
+        }
+        // CloudPlane end
+
         return component == null ? Component.empty() : GsonComponentSerializer.gson().serializer().fromJson(net.minecraft.network.chat.Component.Serializer.toJsonTree(component), Component.class);
     }
 
@@ -193,6 +199,12 @@ public final class PaperAdventure {
     }
 
     public static String asJsonString(final net.minecraft.network.chat.Component component, final Locale locale) {
+        // CloudPlane start
+        if (true) {
+            return asJsonString(asAdventure(component), locale);
+        }
+        // CloudPlane end
+
         if (component instanceof AdventureComponent) {
             return asJsonString(((AdventureComponent) component).adventure, locale);
         }
diff --git a/src/main/java/net/minecraft/world/entity/Entity.java b/src/main/java/net/minecraft/world/entity/Entity.java
index 1b1ef2b4b8441fa3c94b06ebb7713a2428d4a769..6aed3d3cda132a33089a9ae0dd44daa41bf7c861 100644
--- a/src/main/java/net/minecraft/world/entity/Entity.java
+++ b/src/main/java/net/minecraft/world/entity/Entity.java
@@ -2486,7 +2486,7 @@ public abstract class Entity implements Nameable, EntityAccess, CommandSource {
                         String s = nbt.getString("CustomName");
 
                         try {
-                            this.setCustomName(Component.Serializer.fromJson(s));
+                            this.setCustomName(io.papermc.paper.adventure.PaperAdventure.asVanilla(net.kyori.adventure.text.serializer.gson.GsonComponentSerializer.gson().deserialize(s))); // CloudPlane
                         } catch (Exception exception) {
                             Entity.LOGGER.warn("Failed to parse entity custom name {}", s, exception);
                         }
